packages/jest-message-util/src/index.ts

```diff
@@ -383,15 +435,18 @@ export const formatResultsErrors = (
   }
 
   return failedResults
-    .map(({result, content}) => {
-      let {message, stack} = separateMessageFromStack(content);
-      stack = options.noStackTrace
-        ? ''
-        : `${STACK_TRACE_COLOR(
-            formatStackTrace(stack, config, options, testPath),
-          )}\n`;
-
-      message = indentAllLines(message);
+    .map(({result, content, failureDetails}) => {
+      const rootErrorOrStack: Error | string = failureDetails
+        ? types.isNativeError(failureDetails) || failureDetails instanceof Error
+          ? failureDetails // receiving raw errors for jest-circus
+          : typeof failureDetails === 'object' &&
+            'error' in failureDetails &&
+            failureDetails.error &&
+            (types.isNativeError(failureDetails.error) ||
+              failureDetails.error instanceof Error)
+          ? failureDetails.error // receiving instances of FailedAssertion for jest-jasmine
+          : content
+        : content;
```

Comment:
this is hard to read - could you extract to a helper function with `return`s?

Reply:
Definitely, will do